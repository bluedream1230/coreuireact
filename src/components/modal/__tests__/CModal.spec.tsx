import * as React from 'react'
import { render /* ,screen */, fireEvent /* ,waitFor */ } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CModal } from '../../../index'

test('loads and displays CModal component', async () => {
  const { container } = render(<CModal>Test</CModal>)
  expect(container).toMatchSnapshot()
})

test('CModal customize', async () => {
  const { container } = render(
    <CModal
      alignment="center"
      className="bazinga"
      duration={100}
      fullscreen="xl"
      scrollable={true}
      size="xl"
      visible={true}
    >
      Test
    </CModal>,
  )
  expect(container).toMatchSnapshot()
})

test('CModal dialog close on press ESC', async () => {
  jest.useFakeTimers()
  const onDismiss = jest.fn()
  render(
    <CModal onDismiss={onDismiss} visible>
      Test
    </CModal>,
  )
  expect(onDismiss).toHaveBeenCalledTimes(0)
  const modal = document.querySelector('.modal')
  if (modal !== null) {
    fireEvent.keyDown(modal, {
      key: 'Escape',
      code: 'Escape',
      keyCode: 27,
      charCode: 27,
    })
  }
  jest.runAllTimers()
  expect(onDismiss).toHaveBeenCalledTimes(1)
  jest.useRealTimers()
})

test('CModal dialog close on backdrop', async () => {
  jest.useFakeTimers()
  const onDismiss = jest.fn()
  render(
    <CModal onDismiss={onDismiss} visible={true}>
      Test
    </CModal>,
  )
  expect(onDismiss).toHaveBeenCalledTimes(0)
  const backdrop = document.querySelector('.modal-backdrop')
  if (backdrop !== null) {
    fireEvent.click(backdrop)
  }
  jest.runAllTimers()
  expect(onDismiss).toHaveBeenCalledTimes(1)
  jest.useRealTimers()
})
