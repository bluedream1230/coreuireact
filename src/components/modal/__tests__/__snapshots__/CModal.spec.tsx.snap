// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CModal customize 1`] = `
<div>
  <div>
    <div
      class="modal-backdrop show"
    />
    <div
      class="modal fade bazinga show d-block"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-dialog-centered modal-fullscreen-xl-down modal-dialog-scrollable modal-xl"
      >
        <div
          class="modal-content"
        >
          Test
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`loads and displays CModal component 1`] = `
<div>
  <div>
    <div
      class="modal fade"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          Test
        </div>
      </div>
    </div>
  </div>
</div>
`;
